Agents/PubChemAgent/README.md

```diff
@@ -1,151 +1,246 @@
+# Authenticate with the GitHub Container Registry (if not already logged in)
+docker login ghcr.io
+# If prompted, enter your GitHub username and a personal access token with permission and scope to read and write the container repository.
 
-```bash
-docker pull docker.cmclinnovations.com/stack-client:1.6.2
+# Publish the Docker image to the Github Container Registry
+docker image push ghcr.io/theworldavatar/pubchem_agent:1.1.0
 ```
 
 ### Docker deployment
 
-Deploy the dockerised agent by running the following code in the command prompt from the same location where this README is located (ideally, use a bash terminal to avoid potential issues with inconsistent path separators).
+To deploy the Dockerised agent, run the following commands from the directory where this README is located (use a bash terminal to avoid potential issues with inconsistent path separators).
 
 ```bash
-# Deploy the Docker images (production / debug) locally
+# Run the production container locally
 docker-compose -f docker-compose.yml  up
+
+# Run the debug container locally
 docker-compose -f docker-compose.debug.yml  up
 ```
 
-To verify the correct startup of the agent, open the URL address the agent is running on, e.g. <http://localhost:5000> in your browser.
+To verify that the agent has started correctly, open <http://localhost:5000> in your browser.
+
+To retrieve the IRI of a species (e.g., argon), open <http://localhost:5000/query/species?inchi=InChI=1/Ar>. If argon already exists in the namespace, the agent will return its IRI.
+If not, the agent queries PubChem, instantiates the species in the target namespace, and returns the newly created IRI.
+
+If you are able to access this endpoint successfully, it means the agent has been deployed correctly. You can now begin retrieving and instantiating species using their InChIs.
+
+**Important notes:**
+
+The PubChem Agent works fully with any external Blazegraph endpoint, as long as the required environment variables `NAMESPACE`, `QUERY_ENDPOINT` and `UPDATE_ENDPOINT` are correctly set.
+
+If your deployment already relies on a remote Blazegraph instance (for example, a server-hosted namespace), you can safely remove the Blazegraph service definition from both the `docker-compose.yml` and `docker-compose.debug.yml` files:
+
+This means deleting the following blocks:
+
+```yaml
+  blazegraph_os:
+    container_name: "blazegraph_os"
+    image: ghcr.io/cambridge-cares/blazegraph:1.1.0
+    restart: unless-stopped
+    ports:
+      - "8080:8080"
+    volumes:
+      - blazegraph_os_data:/blazegraph
+    networks:
+      - default
+
+volumes:
+  blazegraph_os_data:
+    name: "blazegraph_os_data"
+```
 
 ### Stack Deployment
 
-If you want to spin up this agent as part of a stack, do the following:
+If you want to deploy this agent as part of a TWA stack, follow these steps:
+
+- Set the ```STACK_NAME``` in your ```docker-compose.yml``` file, for example:
+  ```bash
+  STACK_NAME=chemistry_stack
+  ```
```

Comment:
The stack manager does not use docker compose files at all - they are replaced by the service config json files, and the environment variables are used only at runtime and do not affect the build process. I believe you can simply remove the above four lines, given what follows below.